src/extension/background-script/CryptoServices/prepareRecipientDetail.ts

```diff
@@ -19,9 +21,7 @@ export async function prepareRecipientDetail(to: (ProfileIdentifier | GroupIdent
                 await Promise.all(
                     group.members.map((x) => self(x, _, __, { type: 'group', at: new Date(), group: who })),
                 )
-            } else {
-                throw new TypeError('Invalid recipients type')
-            }
+            } else unreachable(who)
```

Comment:
I think it's unnecessary to throw an error with a function call here.

Reply:
In this case, `who` will be a `never` type. The `unreachable` function receives a `never` type so if we miss some case in the future, the typescript will throw error for us. It's a type level hack